app/src/components/v-input.vue

```diff
@@ -110,11 +125,19 @@ const classes = computed(() => [
 ]);
 
 const isStepUpAllowed = computed(() => {
-	return props.disabled === false && (props.max === undefined || parseInt(String(props.modelValue), 10) < props.max);
+	return (
+		props.disabled === false &&
+		(props.max === undefined || parseInt(String(props.modelValue), 10) < props.max) &&
+		new SafeInteger(inputValue.value, props.isBigInt).isMaximum === false
```

Comment:
I'd prefer this to be a function call over a class instantiation for readability (perhaps a static method or wrapper)
```suggestion
		new SafeInteger(inputValue.value, props.isBigInt).isMaximum === false
```